tidy3d/plugins/near2far/near2far.py

```diff
@@ -1,130 +1,483 @@
 """Near field to far field transformation plugin
 """
+from typing import List, Dict
 import numpy as np
-
-from ...constants import C_0, ETA_0
-from ...components.data import SimulationData
+import xarray as xr
+import pydantic
+
+from ...constants import C_0, ETA_0, MICROMETER
+from ...components.data import SimulationData, FieldData
+from ...components.monitor import FieldMonitor
+from ...components.types import Direction, Axis, Size
+from ...components.medium import Medium
+from ...components.validators import assert_plane
 from ...log import SetupError
 
-# TODO: implement new version with simulation.discretize
 
+class Near2FarSurface(pydantic.BaseModel):
+    """Data structure to store surface monitor data with associated surface current densities."""
+
+    monitor: FieldMonitor = pydantic.Field(
+        ...,
+        title="Field monitor",
+        description=
+        "Object of :class:`.FieldMonitor` on which near fields will be sampled and integrated.",
+    )
+
+    normal_dir: Direction = pydantic.Field(
+        '+',
+        title="Normal vector orientation",
+        description=
+        ":class:`.Direction` of the surface monitor's normal vector w.r.t.\
+ the positive x, y or z unit vectors. Must be one of '+' or '-'.",
+    )
+
+    size: Size = pydantic.Field(
+        None,
+        title="Size",
+        description="Size in x, y, and z directions.",
+        units=MICROMETER,
+    )
+
+    @property
+    def axis(self) -> Axis:
+        """Returns the :class:`.Axis` normal to this surface."""
+        # assume that the monitor's axis is in the direction where the monitor is thinnest
+        return np.argmin(self.monitor.size)
+
+    @pydantic.validator("size", always=True)
```

Comment:
see above, just reference `self.monitor.size` whenever you need it and remove this